manager/logbroker/broker.go

```diff
@@ -23,25 +24,30 @@ var (
 	errNotRunning     = errors.New("broker is not running")
 )
 
-// LogBroker coordinates log subscriptions to services and tasks. Çlients can
+// LogBroker coordinates log subscriptions to services and tasks. Clients can
 // publish and subscribe to logs channels.
 //
 // Log subscriptions are pushed to the work nodes by creating log subscsription
 // tasks. As such, the LogBroker also acts as an orchestrator of these tasks.
 type LogBroker struct {
-	mu                sync.RWMutex
-	logQueue          *watch.Queue
-	subscriptionQueue *watch.Queue
+	mu                       sync.RWMutex
+	logQueue                 *watch.Queue
+	newSubscriptionsQueue    *watch.Queue
+	changeSubscriptionsQueue *watch.Queue
```

Comment:
I feel like it might be simpler to combine the two queues. I think a new subscription that matches a node and a subscription that has changed so it now matches a node are so similar that it makes sense to handle them the same way. The combined queue could have `*matcher` objects, like `newSubscriptionsQueue`, and `watchSubscriptions` would call `Contains` on every item it receives.